test/test_field_change.py

```diff
@@ -0,0 +1,208 @@
+import unittest
+import os.path
+
+from tableaudocumentapi import Datasource
+import xml.etree.ElementTree as ET
+
+
+TEST_ASSET_DIR = os.path.join(
+    os.path.dirname(__file__),
+    'assets'
+)
+TEST_TDS_FILE = os.path.join(
+    TEST_ASSET_DIR,
+    'field_change_test.tds'
+)
+TEST_TDS_FILE_OUTPUT = os.path.join(
+    TEST_ASSET_DIR,
+    'field_change_test_output.tds'
+)
+
+MESSAGES = {
+    'test_change_values1': 'Value has not changed when altering values for {}.',
+    'test_change_values2': 'XML-Structure has not changed when altering values for {}.',
+    'test_change_valuesFail1': 'Value has changed when submitting the wrong value for {}.',
+    'test_change_valuesFail2': 'XML-Structure has changed when submitting the wrong value for {}.',
+    'test_change_aliases1': 'XML-Structure has not changed when altering aliases for {}.',
+    'test_change_aliases2': 'Values have not changed when altering aliases for {}.'
+
+}
+
+NEW_VALUES = {
+    'caption': 'testcaption',
+    'alias': 'testalias',
+    'datatype': 'boolean',
+    'role': 'measure',
+    'type': 'ordinal'
+}
+
+WRONG_VALUES = {
+    'datatype': 'boolani',
+    'role': 'messhure',
+    'type': 'gordinol'
+}
+
+ALIASES = {
+    'one': 'two',
+    'three': 'four',
+    'five': 'six'
+}
+
+
+class TestFieldChange(unittest.TestCase):
+
+    def setUp(self):
+        self.tds = Datasource.from_file(TEST_TDS_FILE)
+
+    def current_hash(self):
+        """ Return a hash of the current state of the XML.
+
+        Allows us to easily identify whether the underlying XML-structure
+        of a TDS-file has actually changed. Avoids false positives if,
+        for example, a fields value has changed but the XML hasn't.
+        """
+        return hash(ET.tostring(self.tds._datasourceTree.getroot()))
+
+    def test_change_values(self):
```

Comment:
There are many similarities between the code in this test and the next one.  Is there a way to extract the mostly common code and replace the differences with arguments to the new function?